### ISSUE TITLE
Better message if code lenses fail with an error

### ISSUE THREAD

**Issue by michaelpj** [emacs-lsp/lsp-mode#2250]
[This issue](https://github.com/emacs-lsp/lsp-haskell/issues/89) was I believe originally raised because the user got a message saying they should raise an issue asking for an implementation.

However, the actual problem is that the server sent back an error because the execution of the lens failed. I'm not 100% sure what's going on here, but it seems like we should be able to tell in this case that it's not a "missing implementation", but rather the server has told us it can do this code lens, but then fails when we try - i.e. it's a bug in the server.

**Comment by kiennq** [comment]
> I'm not 100% sure what's going on here, but it seems like we should be able to tell in this case that it's not a "missing implementation", but rather the server has told us it can do this code lens, but then fails when we try - i.e. it's a bug in the server.

I'm not really sure that this should be bug in server for all the cases.
For example, `rust-analyzer` has `rust-analyzer.debugSingle` and `rust-analyzer.runSingle` as code lens and expect client to implement it instead of server.

We should just notify user to open bug against the repo and after triage, either open bug for server or implement it in client.

**Comment by nbfalcon** [comment]
I agree that we need better error messages here. It would be great if we could get access to the original error's :code in `lsp-request` or `lsp-request-async`. @yyoncho why does the :error-handler in `lsp-request-async` not get the original error, as sent from the server, but instead only a description of it?

**Comment by yyoncho** [comment]
@michaelpj can you provide what exactly is the result from the language server call?

**Comment by michaelpj** [comment]
I didn't reproduce it myself, but this is what the user showed:
```
(error "BadDependency "GhcSessionDeps"")
LSP :: Please open an issue in lsp-mode for implementing `21444:eval:evalCommand'.
```

**Comment by michaelpj** [comment]
i.e. it looks like we get *something* back from the server. Not very descriptive, but that's the server implementor's problem! We should ideally just be able to say: "Your server gave this error when you tried to do that. Go show it to them."

**Comment by yyoncho** [comment]
Ok, so the actual error pops up. It seems to me that we should drop the message and let the original error pop-up. We cannot distinguish between actual valid failure and missing implementation.

**Comment by kiennq** [comment]
Here is current code for this message
``` elisp
(lsp--error "Please open an issue in lsp-mode for implementing `%s'.\n\n%S"
                 command err)
```
I think the error is also being displayed now. So we can help route the issue to the language server.
Or we can reword to not mention about open new issue.
Without this kind of message, it would be hard actually know which command is causing problem in case of missing implementation.

**Comment by alanz** [comment]
I think it would help if the lsp-mode logging include explicit error messages.

This is the hls log
```
2021-01-16 18:37:29.425894601 [ThreadId 7] - <--2--{"jsonrpc":"2.0","params":{"value":{"kind":"end"},"token":1},"method":"$/progress"}
2021-01-16 18:37:29.42614394 [ThreadId 7] - <--2--{"error":{"code":-32603,"message":"attempting to use module ‘fake_uid:Main’ (/tmp/fff/test.hs) which is not loaded\n"},"jsonrpc":"2.0","id":305}
2021-01-16 18:37:29.680290767 [ThreadId 5] - ---> {"jsonrpc":"2.0","method":"textDocument/codeAction","params":{"textDocument":{"uri":"file:///tmp/fff/test.hs"},"range":{"start":{"line":3,"character":0},"end":{"line":3,"character":0}},"context":{"diagnostics":[]}},"id":306}
```

And this is the corresponding lsp-mode log

*lsp-log:xxxx"
```
[Trace - 06:37:29 ] Received notification '$/progress'.
Params: {
  "token": 1,
  "value": {
    "kind": "end"
  }
}


[Trace - 06:37:29 ] Received response 'workspace/executeCommand - (305)' in 24ms.
Result: null


[Trace - 06:37:29 ] Sending request 'textDocument/codeAction - (306)'.
```

The error detail is discarded, giving `Result:null` instead.

**Comment by yyoncho** [comment]
@alanz this is lsp-mode bug.
